experimental/IntersectionTheory/src/kontsevich.jl

```diff
@@ -0,0 +1,460 @@
+        for f in Aut
+          f_edge_c = sort!([findfirst(==(sort!([f(src(es[i])), f(dst(es[i]))])), ess) for i in edge_c])
+          if f_edge_c == edge_c aut += 1 end
+          filter!(!=(f_edge_c), edge_configs)
+        end
+        mul_g = MultiGraph(_copy(g), Dict([e => multi[i] for (i, e) in enumerate(es)]), aut*prod(multi))
+        push!(ans, mul_g)
+      end
+    end
+  end
+  ans
+end
+
+# ###############################################################################
+# # 
+# # low-level in-place arithmetic operators
+# # should probably be added into Nemo upstream
+# # 
+# function set!(x::fmpz, y::Int)
+#   ccall((:fmpz_set_si, Nemo.libflint), Nothing, (Ref{fmpz}, Int), x, y)
+#   return x
+# end
+# function set!(x::QQ, y::QQ)
+#   ccall((:fmpz_set, Nemo.libflint), Nothing, (Ref{QQ}, Ref{QQ}), x, y)
+#   return x
+# end
+function Nemo.set!(x::QQFieldElem, n::Int, d::Int)
+  ccall((:fmpq_set_si, Nemo.libflint), Nothing, (Ref{QQFieldElem}, Int, UInt), x, n, UInt(d))
+  return x
+end
+# function set!(x::QQ, n::fmpz, d::fmpz)
+#   ccall((:QQ_set_fmpz_frac, Nemo.libflint), Nothing, (Ref{QQ}, Ref{fmpz}, Ref{fmpz}), x, n, d)
+#   return x
+# end
+# function Nemo.mul!(z::QQ, x::QQ, y::fmpz)
+#   ccall((:QQ_mul_fmpz, Nemo.libflint), Nothing, (Ref{QQ}, Ref{QQ}, Ref{fmpz}), z, x, y)
+#   return z
+# end
+# function Nemo.mul!(z::QQ, x::QQ, y::Int)
+#   ccall((:QQ_mul_si, Nemo.libflint), Nothing, (Ref{QQ}, Ref{QQ}, Int), z, x, y)
+#   return z
+# end
+# function Nemo.sub!(z::QQ, x::QQ, y::Int)
+#   ccall((:QQ_sub_si, Nemo.libflint), Nothing, (Ref{QQ}, Ref{QQ}, Int), z, x, y)
+#   return z
+# end
+# function Nemo.inv!(x::QQ, y::QQ)
+#   ccall((:QQ_inv, Nemo.libflint), Nothing, (Ref{QQ}, Ref{QQ}), x, y)
+#   return x
+# end
+# function pow!(c::QQ, a::QQ, b::Int)
+#   iszero(a) && b < 0 && throw(DivideError())
+#   ccall((:QQ_pow_si, Nemo.libflint), Nothing, (Ref{QQ}, Ref{QQ}, Int), c, a, b)
+#   return c
+# end
+# function div!(z::QQ, x::QQ, y::QQ)
+#   iszero(y) && throw(DivideError())
+#   ccall((:QQ_div, Nemo.libflint), Nothing, (Ref{QQ}, Ref{QQ}, Ref{QQ}), z, x, y)
+#   return z
+# end
+# 
+# muleq!(x::QQ, y::QQ) = mul!(x, x, y)
+# muleq!(x::QQ, y::fmpz) = mul!(x, x, y)
+# muleq!(x::QQ, y::Int) = mul!(x, x, y)
+# subeq!(x::QQ, y::Int) = Nemo.sub!(x, x, y)
+# poweq!(x::QQ, y::Int) = pow!(x, x, y)
+# diveq!(x::QQ, y::QQ) = div!(x, x, y)
+# Nemo.inv!(x::QQ) = Nemo.inv!(x, x)
+
+###############################################################################
+# 
+# miscellaneous functions 
+# 
+
+function _part(n::T, rem::T, k::T, m::T, part::Vector{T}, ans::Vector{Generic.Partition{T}}) where T <: Integer
+  rem == 0 && (push!(ans, Generic.Partition(n, part[1:end-k], false)); return)
+  k <= 0 && return
+  for v in min(rem, m):-T(1):T(1)
+    part[end-k+1] = v
+    _part(n, rem-v, k-T(1), v, part, ans)
+  end
+end
+# partitions of n with at most k numbers each <= m
+function _partitions(n::T, k::T=n, m::T=n) where T <: Integer
```

Comment:
If I understand this comment correctly, the function does the same as:
```
collect(Iterators.flatten(partitions(n, i, 1, m) for i in 1:k))
```

Reply:
Probably, but the type is different and we don't want to think about this right now. (We will clean it up "later")